MetaMask/metamask-extension
# Fix `TRANSACTION_MULTICHAIN` flow

## **Description**

Makes necessary changes to get the multichain transaction flow semi-functioning

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27232?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**
NETWORK A should be the chain you want to transact with for a specific account
NETWORK B should be a different chain from NETWORK A, but one that you also have a balance on for the specific account (so that the UI does not disable send/continue buttons due to insufficient balance)


1. `TRANSACTION_MULTICHAIN=1 yarn start`
2. Connect a dapp
3. Click on the metamask icon to open the popup
4. Change the globally selected network to `NETWORK A` which will also change it for the dapp
6. Open a new tab and click on the metamask icon to open the popup
7. Change the globally selected network to `NETWORK B`, but this time it won't change it for the dapp
8. Open the dapp tab again, DO NOT click the metamask icon again (or the globally selected network will be pointed at the dapp selected network, which we don't want)
9. Verify the dapp is still on the right network `window.ethereum.chainId`
10. Initiate a send transaction from the dapp
11. 

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
